clients/tabby-agent/src/TabbyAgent.ts

```diff
@@ -136,6 +136,7 @@ export class TabbyAgent extends EventEmitter implements Agent {
     }
 
     if (!this.api || isServerConnectionChanged) {
+      this.changeStatus("notInitialized");
```

Comment:
@icycodes this is required to didChangeStatus will be triggered.

Before this change, if user switching config A to config B, and each config are legit, there won't be another `ready` status being triggered due to event deduping check.

Reply:
LGTM